### DISCUSSION TITLE
How to update the OS?

### DISCUSSION THREAD

**Discussion by chetstone** [elementary/appcenter#2218] [Q&A] [answered]
I'm just trying Elementary OS on a 2012 Macbook Air. I installed version 7.1, which is about a year old. I assume there must be some updates. In fact, one of your blog entries says:

> As always, pop open AppCenter on elementary OS 7 and hit “Update All” to get all these updates plus your regular security, bug fix, and translation updates.

I cannot find "Update All".

When I open AppCenter, a popup in the lower right corner gives several messages about checking for updates, such as "Getting download size". The AppCenter icon in the dock has a badge indicating 11 (updates, I assume?). 
When the popup is finished, when I click the Orange button at the top of the screen, the main frame says "Checking for Updates", as in the following screenshot. This screen never changes, and I cannot find an "Update All" button anywhere.
![Screenshot from 2024-10-12 11 15 02](https://github.com/user-attachments/assets/c9910f91-4a18-4500-837d-b59c1bba5c09)

Since nobody else seems to have asked this question, I am feeling really dumb.

**Comment by jeremypw** [comment]
No, you are not dumb. There seems to be a problem with the app - it should show a list of available updates after a few seconds and also an "Update All" button.  It does on my system (but that is not a Mac).  Not sure what the exact problem is, but as a workround you could try opening the Terminal app (after closing the AppCenter) and run `sudo apt update`.  This will tell what needs updating.  You can then run `sudo apt upgrade` to actually install the updates.  To update any flatpaks you have run `flatpak update` and respond to the prompts.  I usually do it this way tbh as you get more information and it can be quicker.  I use AppCenter for search/install/uninstall mostly. 

Maybe someone with elementary on Mac can add to this.

**Comment by danirabbit** [comment, marked as answer]
So sorry to hear that you're feeling down on yourself! This was a software bug that was fixed recently where sometimes AppCenter would get stuck on the loading screen.

Jeremy is right that probably the best way to get the update that fixes this issue is by opening the Terminal app and typing in the following command, and pressing enter:
`sudo apt install io.elementary.appcenter`

This will ask you for your password and then install the latest version of AppCenter. It may ask you to confirm by entering `y` and pressing enter

If an update for AppCenter is not found or this still doesn't resolve the issue you may have to type the following command in Terminal and pressing enter. This command will search for additional updates:

`sudo apt update`

After this update is installed, the issue with the updates page in AppCenter should be resolved and you can update normally from there
